app/views/users/_form.html.erb

```diff
@@ -26,6 +26,9 @@
       <%= text_f f, :firstname %>
       <%= text_f f, :lastname %>
       <%= text_f f, :mail %>
+      <% unless @user == User.current %>
```

Comment:
This correctly hides the checkbox in UI. It is still possible to disable my own account when posting to `update` in both API and UI controllers.